### PR TITLE
update delete operation for statefulset pod

### PR DESCRIPTION
**Pull request by hongzhen-ma**
#### What type of this PR
- Bug fixes
####
1、statefulset pod is running
2、change subnet for ns which statefulset pod belongs to
3、delete statefulset pod, and it can not rerun successfully

### CODE REVIEW

File: pkg/controller/pod.go [hunk 1]
```diff
@@ -1210,3 +1215,21 @@ func (c *Controller) deleteAttachmentNetWorkIP(pod *v1.Pod) error {
 	}
 	return nil
 }
+
+func appendCheckStatefulSetPodToDel(c *Controller, pod *v1.Pod) bool {
+	// subnet for ns has been changed, and statefulset pod's ip is not in the range of subnet's cidr anymore
+	podNs, _ := c.namespacesLister.Get(pod.Namespace)
```
Comment (oilbeater) [inline pkg/controller/pod.go:1221]: should check error herre

File: pkg/controller/pod.go [hunk 2]
```diff
@@ -1210,3 +1215,21 @@ func (c *Controller) deleteAttachmentNetWorkIP(pod *v1.Pod) error {
...
+	if podNs.Annotations[util.LogicalSwitchAnnotation] != "" && pod.Annotations[util.LogicalSwitchAnnotation] != "" && strings.TrimSpace(podNs.Annotations[util.LogicalSwitchAnnotation]) != strings.TrimSpace(pod.Annotations[util.LogicalSwitchAnnotation]) {
+		klog.Infof("ns %s annotation subnet is %s, which is inconstant with subnet for pod %s, delete pod", pod.Namespace, podNs.Annotations[util.LogicalSwitchAnnotation], pod.Name)
+		return true
+	}
+
+	// subnet cidr has been changed, and statefulset pod's ip is not in the range of subnet's cidr anymore
+	podSubnet, _ := c.subnetsLister.Get(strings.TrimSpace(pod.Annotations[util.LogicalSwitchAnnotation]))
```
Comment (oilbeater) [inline pkg/controller/pod.go:1228]: same as above

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/controller/pod.go (hunk 1 above)
Comment (hongzhen-ma) [reply pkg/controller/pod.go]: modified

File: pkg/controller/pod.go (hunk 2 above)
Comment (hongzhen-ma) [reply pkg/controller/pod.go]: modified

**Review by oilbeater — APPROVED** [review]

**Pull request merged**